packages/tools/examples/videoColor/index.ts

```diff
@@ -0,0 +1,295 @@
+
+    // toggle the title
+    const button = document.getElementById('play');
+    if (button.innerText === 'pause') {
+      button.innerText = 'play';
+    } else {
+      button.innerText = 'pause';
+    }
+  },
+});
+
+const whiteValues = [
+  [255, 255, 255],
+  [180, 255, 255],
+  [255, 180, 255],
+  [255, 255, 180],
+  [255, 180, 180],
+  [180, 255, 180],
+  [180, 180, 255],
+];
+let currentWhite = 0;
+
+addButtonToToolbar({
+  id: 'Colour Correct',
+  title: 'Colour: 255,255,255',
+  onClick() {
+    currentWhite = (1 + currentWhite) % whiteValues.length;
+    const white = whiteValues[currentWhite];
+    viewport.setAverageWhite(white);
+    document.getElementById('Colour Correct').innerText = `Color: ${white.join(
+      ','
+    )}`;
+  },
+});
+
+/**
+ * One possible average white function showing how the imageData is used
+ * to get average white information.
+ */
+function getAverageWhite(scalarData) {
+  const maxValues = [0, 0, 0];
+  for (let i = 0; i < scalarData.length; i += 4) {
+    const r = scalarData[i];
+    const g = scalarData[i + 1];
+    const b = scalarData[i + 2];
+    maxValues[0] = Math.max(r, maxValues[0]);
+    maxValues[1] = Math.max(g, maxValues[1]);
+    maxValues[2] = Math.max(b, maxValues[2]);
+  }
+  return maxValues;
+}
+
+addButtonToToolbar({
+  id: 'Avg Colour Correct',
+  title: 'Avg Colour Correct',
+  onClick() {
+    const white = getAverageWhite(
+      viewport.getImageData().imageData.getScalarData()
+    );
+    console.log('White=', white);
+    viewport.setAverageWhite(white);
+    document.getElementById(
+      'Colour Correct'
+    ).innerText = `Avg Color: ${white.join(',')}`;
+    currentWhite = -1;
+  },
+});
+
+const wl = (windowWidth, windowCenter) => ({ windowWidth, windowCenter });
+const windowLevels = [
+  wl(256, 128),
+  wl(255, 127.5),
+  wl(192, 96),
+  wl(192, 128),
+  wl(192, 160),
+];
+
+const toolbar = document.getElementById('demo-toolbar');
+const windowLevelNames = windowLevels.map(
+  ({ windowWidth, windowCenter }) => `W:${windowWidth}/C:${windowCenter}`
+);
+addDropdownToToolbar({
+  options: {
+    values: windowLevelNames,
+    defaultValue: windowLevelNames[0],
+    id: 'windowLevel',
+  },
+  onSelectedValueChange: (newWLText) => {
+    const index = windowLevelNames.indexOf(newWLText) % windowLevels.length;
+    const newWL = windowLevels[index];
+    viewport.setWindowLevel(newWL.windowWidth, newWL.windowCenter);
+  },
+});
+
+const urlParams = new URLSearchParams(window.location.search);
+const useLocal = urlParams.get('useLocal') === 'true';
+
+/**
+ * Runs the demo
+ */
+async function run() {
+  // Init Cornerstone and related libraries
+  await initDemo();
+
+  // Get Cornerstone imageIds and fetch metadata into RAM
+  const imageIds = await createImageIdsAndCacheMetaData({
+    StudyInstanceUID: '2.25.96975534054447904995905761963464388233',
+    SeriesInstanceUID: '2.25.15054212212536476297201250326674987992',
+    wadoRsRoot: useLocal
```

Comment:
pull main so that you can get the new local approach

Reply:
It isn't in main yet since it is part of HTJ2K, which hasn't been merged.  Prefer not to have it added twice and it is easy to fix once htj2k is merged.